pkg/engine/variables/vars.go

```diff
@@ -18,7 +18,8 @@ import (
 )
 
 var RegexVariables = regexp.MustCompile(`\{\{[^{}]*\}\}`)
-var RegexReferences = regexp.MustCompile(`\$\(.[^\ ]*\)`)
+var RegexReferences = regexp.MustCompile(`^\$\(.[^\ ]*\)|[^\\]\$\(.[^\ ]*\)`)
+var RegexEscpReferences = regexp.MustCompile(`\\\$\(.[^\ ]*\)`)
```

Comment:
Can you add a brief description to the new regex?

Reply:
Please mark the comment as resolved once you addressed it. It's easier for the reviewer to track the changes.